test/extended/openstack/bz_2073398.go

```diff
@@ -136,6 +142,42 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack] Bugfix", func() {
 				}
 			}
 			o.Expect(portID).To(o.Equal(""))
+			bogusMachines, bogusMachinesNames, err := getMachinesByPrefix(ms.Name, ctx, dc)
+			o.Expect(err).NotTo(o.HaveOccurred(), "Failed to get machines")
+			o.Expect(len(bogusMachines)).To(o.Equal(2), "Machines %v should have been deleted", bogusMachinesNames)
 		})
 	})
 })
+
+func getMachinesByPrefix(prefix string, ctx context.Context, dc dynamic.Interface) ([]objx.Map, []string, error) {
+	var machinesList []objx.Map
+	var machinesNames []string
+
+	machines, err := machines.List(ctx, dc)
+
+	for _, machine := range machines {
+		machineName := machine.Get("metadata.name").String()
+		if strings.HasPrefix(machineName, prefix) {
+			machinesList = append(machinesList, machine)
+			machinesNames = append(machinesNames, machineName)
+		}
+	}
+	return machinesList, machinesNames, err
+}
+
+// Wait until N machines with a specific prefix exist
+func waitUntilNMachinesPrefix(prefix string, numMachines int, ctx context.Context, dc dynamic.Interface) error {
```

Comment:
You should accept a context here and set a timeout on the calling side

Reply:
I changed that to a different function